py/legacyanalysis/sersic-agn.py

```diff
@@ -0,0 +1,269 @@
+    pass
+
+def flux_string(br):
+    s = []
+    for band in 'grz':
+        flux = br.getFlux(band)
+        if flux <= 0:
+            s.append('%s=(%.2f nmy)' % (band, flux))
+        else:
+            s.append('%s=%.2f' % (band, NanoMaggies.nanomaggiesToMag(flux)))
+    s = ', '.join(s)
+    return s
+
+def showmods(tims,mods):
+    #plt.figure(figsize=(10,6))
+    cols = len(tims) // 3
+    panels = [((tim.getImage()-mod)*tim.getInvError()) for tim,mod in list(zip(tims, mods))]
+    h = min([p.shape[0] for p in panels])
+    w = min([p.shape[1] for p in panels])
+    panels = [p[:h,:w] for p in panels]
+    stack = []
+    while len(panels):
+        stack.append(np.hstack(panels[:cols]))
+        panels = panels[cols:]
+    stack = np.vstack((stack))
+    plt.imshow(stack, vmin=-2, vmax=2, **ima)
+    plt.xticks(w * (0.5 + np.arange(cols)), np.arange(cols))
+    plt.yticks(h * (0.5 + np.arange(3)), ['g','r','z'])
+
+def showresid(tims,mods,wcs,bands='grz'):
+    co,_ = quick_coadds(tims, bands, wcs)
+    comod,_ = quick_coadds(tims, bands, wcs, images=mods)    
+    plt.imshow(np.flipud(get_rgb([i-m for i,m in zip(co,comod)], bands)))
+    plt.xticks([]); plt.yticks([])
+    return co,comod
+    
+def showboth(tims,mods,wcs,bands):
+    fig = plt.figure(num=1, figsize=(14,6), constrained_layout=True)
+    fig.clf()
+    gs = fig.add_gridspec(3, 4)
+    ax = fig.add_subplot(gs[:, :3])
+
+    cols = len(tims) // 3
+    rows = 3
+    panels = [((tim.getImage()-mod)*tim.getInvError()) for tim,mod in list(zip(tims, mods))]
+    h = min([p.shape[0] for p in panels])
+    w = min([p.shape[1] for p in panels])
+    panels = [p[:h,:w] for p in panels]
+    stack = []
+    #hzero = np.zeros((h,1))
+    while len(panels):
+        # hs = [hzero]
+        # for p in panels[:cols]:
+        #     hs.extend([p, hzero])
+        # hs = np.hstack(hs)
+        # hh,hw = hs.shape
+        # if len(stack) == 0:
+        #     wzero = np.zeros((1,hw))
+        #     stack.append(wzero)
+        # stack.append(hs)
+        # stack.append(wzero)
+        stack.append(np.hstack(panels[:cols]))
+        panels = panels[cols:]
+    stack = np.vstack((stack))
+    ax.imshow(stack, vmin=-2, vmax=2, **ima)
+    xl,yl = ax.get_xlim(), ax.get_ylim()
+    #a = ax.get_axis()
+    for c in range(cols+1):
+        plt.axvline(c * w, color='k')
+    for r in range(rows+1):
+        plt.axhline(r * h, color='k')
+    #ax.set_axis(a)
+    ax.set_xlim(xl)
+    ax.set_ylim(yl)
+    #xl,yl = ax.get_xlim(), ax.get_ylim()
+    ax.set_xticks(w * (0.5 + np.arange(cols)))
+    ax.set_xticklabels(np.arange(cols))
+    ax.set_yticks(h * (0.5 + np.arange(3)))
+    ax.set_yticklabels(['g','r','z'])
+
+    ax = fig.add_subplot(gs[2, 3])
+    co,comod = showresid(tims, mods, wcs)
+    ax = fig.add_subplot(gs[0, 3])
+    plt.imshow(np.flipud(get_rgb(co, bands)))
+    plt.xticks([]); plt.yticks([])
+    ax = fig.add_subplot(gs[1, 3])
+    plt.imshow(np.flipud(get_rgb(comod, bands)))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['get_rgb' may be undefined, or defined from star imports: legacypipe.survey, tractor, tractor.devagn, tractor.galaxy](https://app.codacy.com/app/dstndstn/legacypipe/pullRequest?prid=4730106)